src/redux/preview.js

```diff
@@ -125,6 +160,124 @@ module.exports.getCreditInfo = id => (dispatch => {
+        }
+        dispatch(module.exports.setFetchStatus('faved', module.exports.Status.FETCHED));
+        dispatch(module.exports.setFaved(body.userFavorite));
+    });
+});
+
+module.exports.setFavedStatus = (faved, id, username, token) => (dispatch => {
+    if (faved) {
+        api({
+            uri: `/projects/${id}/favorites/user/${username}`,
+            authentication: token,
+            method: 'POST'
+        }, (err, body) => {
+            if (err) {
+                dispatch(module.exports.setError(err));
+                return;
+            }
+            if (typeof body === 'undefined') {
+                dispatch(module.exports.setError('Set favorites returned no data'));
+                return;
+            }
+            dispatch(module.exports.setFetchStatus('faved', module.exports.Status.FETCHED));
+            dispatch(module.exports.setFaved(body.userFavorite));
+        });
+    } else {
+        api({
+            uri: `/projects/${id}/favorites/user/${username}`,
+            authentication: token,
+            method: 'DELETE'
+        }, (err, body) => {
+            if (err) {
+                dispatch(module.exports.setError(err));
+                return;
+            }
+            if (typeof body === 'undefined') {
+                dispatch(module.exports.setError('Set favorites returned no data'));
+                return;
+            }
+            dispatch(module.exports.setFetchStatus('faved', module.exports.Status.FETCHED));
+            dispatch(module.exports.setFaved(false));
+        });
+    }
+});
+
+module.exports.getLovedStatus = (id, username, token) => (dispatch => {
+    dispatch(module.exports.setFetchStatus('loved', module.exports.Status.FETCHING));
+    api({
+        uri: `/projects/${id}/loves/user/${username}`,
+        authentication: token
+    }, (err, body) => {
+        if (err) {
+            dispatch(module.exports.setFetchStatus('loved', module.exports.Status.ERROR));
+            dispatch(module.exports.setError(err));
+            return;
+        }
+        if (typeof body === 'undefined') {
+            dispatch(module.exports.setFetchStatus('loved', module.exports.Status.ERROR));
+            dispatch(module.exports.setError('No loved info'));
+            return;
+        }
+        dispatch(module.exports.setFetchStatus('loved', module.exports.Status.FETCHED));
+        dispatch(module.exports.setLoved(body.userLove));
+    });
+});
+
+module.exports.setLovedStatus = (loved, id, username, token) => (dispatch => {
+    if (loved) {
+        api({
+            uri: `/projects/${id}/loves/user/${username}`,
+            authentication: token,
+            method: 'POST'
+        }, (err, body) => {
+            if (err) {
+                dispatch(module.exports.setError(err));
+                return;
+            }
+            if (typeof body === 'undefined') {
+                dispatch(module.exports.setError('Set loved returned no data'));
+                return;
```

Comment:
Should this be an error? If someone is loving something they already loved, would this be expected?

Reply:
I believe in all cases API returns an object `{userLove: true|false}` as the body.

So, if body is undefined probably something went wrong.